### PR TITLE
Wdu 48 vehicle store implementation

### PR DESCRIPTION
**Pull request by MicheleDonati**
Implemented the vehicle store implementation.

### CODE REVIEW

File: services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        return null;
+    }
+
+    private void checkDuplicatesAndWriteOnVehiclesDb(List<Vehicle> vehicles, ObjectMapper mapper) {
+        if(thereAreNotDuplicates(vehicles)) {
+            try {
+                ClassLoader classLoader = getClass().getClassLoader();
+                File file = new File(classLoader.getResource("vehicles.json").getFile());
+                mapper.writeValue(file, vehicles);
+                String jsonInString = mapper.writeValueAsString(vehicles);
```
Comment (mbaldassarri) [inline services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java]: why you transform the vehicles list to json two times, one raw and one in pretty format with a log in the middle? Sounds like some tests have been left here.

File: services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    }
+
+    private boolean thereAreNotDuplicates(List<Vehicle> vehicles) {
+        for (int j = 0; j < vehicles.size(); j++) {
+            String currentCarLicencePlate = vehicles.get(j).getCarLicencePlate();
+            for (int i = 0; i < vehicles.size(); i++) {
+                if (j != i
+                        && i < vehicles.size() - 1
+                        && vehicles.get(i).getCarLicencePlate().equals(currentCarLicencePlate)) {
+                    Log.log("There are two duplicates car licence plate: "
```
Comment (mbaldassarri) [inline services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java]: same question as above. Logs are good to understand what happens, but in production environment they're useless and make the method less easily readable. Are there really needed here?

File: services/authentication/src/main/java/com/wedriveu/services/authentication/entity/UserStoreImpl.java [hunk 3]
```diff
@@ -71,10 +75,14 @@ private void writeJSonUsersFile(ArrayList<User> userListToJSon){
         }
     }
 
-    private User getUserRequestedCheckingUsersList(User[] users, String username){
+    private User getUserRequestedCheckingUsersList(List<User> users, String username){
         for (User user : users) {
             if(user.getUsername().equals(username)) {
-                Log.log("Login success! " + "Username: " + user.getUsername() + ", Password: " + user.getPassword());
+                Log.log("Login success! " +
```
Comment (MicheleDonati) [inline services/authentication/src/main/java/com/wedriveu/services/authentication/entity/UserStoreImpl.java]: I think that is useful for debugging. Maybe, unless we merge all the components and we're sure that they work perfectly, we can use these logs.

File: services/authentication/src/main/java/com/wedriveu/services/authentication/entity/UserStoreImpl.java [hunk 4]
```diff
@@ -36,7 +39,8 @@ public User getUser(String username) {
         ObjectMapper mapper = new ObjectMapper();
 
         try {
-            User[] users= mapper.readValue(new File(Constants.USERS_DATABASE_PATH), User[].class);
+            List<User> users =
+                    mapper.readValue(new File(Constants.USERS_DATABASE_PATH), new TypeReference<List<User>>(){});
             return getUserRequestedCheckingUsersList(users, username);
```
Comment (MicheleDonati) [inline services/authentication/src/main/java/com/wedriveu/services/authentication/entity/UserStoreImpl.java]: Ok!

File: services/vehicle/src/main/java/Main.java [hunk 5]
```diff
@@ -8,14 +8,41 @@
...
+import java.util.List;
 
 public class Main {
 
     public static void main(String[] args) throws IOException {
         VehicleStoreImpl obj = new VehicleStoreImpl();
-        obj.mapEntityToJson();
+        obj.createVehiclesFile();
         Vehicle me;
         me = obj.getVehicle("MACCHINA1");
```
Comment (MicheleDonati) [inline services/vehicle/src/main/java/Main.java]: Fixed!

File: services/vehicle/src/main/java/com/wedriveu/services/vehicle/entity/VehicleStoreImpl.java [hunk 6]
```diff
@@ -12,24 +13,25 @@
...
 import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 
 public class VehicleStoreImpl implements VehicleStore {
 
     @Override
-    public void mapEntityToJson() {
-        Vehicle vehicle = createDummyObject("MACCHINA1",
+    public void createVehiclesFile() {
```
Comment (MicheleDonati) [inline services/vehicle/src/main/java/com/wedriveu/services/vehicle/entity/VehicleStoreImpl.java]: This method is the one who create the vehicles file at the start of our service. Unless we have the dynamic configuration file that adds vehicles at will, we must create the file first, with this.

File: services/vehicle/src/main/java/com/wedriveu/services/vehicle/entity/VehicleStoreImpl.java [hunk 7]
```diff
@@ -39,64 +41,171 @@ public void mapEntityToJson() {
...
+                Log.log(jsonInString);
+                jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(vehicles);
+                Log.log(jsonInString);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private boolean thereAreNotDuplicates(List<Vehicle> vehicles) {
```
Comment (MicheleDonati) [inline services/vehicle/src/main/java/com/wedriveu/services/vehicle/entity/VehicleStoreImpl.java]: Grammatically is more correct like this, but i changed it, as you prefer :) 
I'm ok in any case.

File: services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java (hunk 2 above)
Comment (MicheleDonati) [inline services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java]: My answer is the same as above. Unless we have all the components entirely merged and working perfectly i prefer that this logs remains.

File: services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java (hunk 1 above)
Comment (MicheleDonati) [inline services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java]: The writeValue method write the vehicles list in the file.json. The writeValueAsString method gives the content of the list in string so we can log it and check what the file contains.

**Pull request merged**